src/index.ts

```diff
@@ -743,6 +776,166 @@ export type ExpectTypeOf<Actual, Options extends {positive: boolean}> = Options[
+          (...args: A5) => R5,
+          (...args: A6) => R6,
+          (...args: A7) => R7,
+          (...args: A8) => R8,
+        ]
+      : FunctionType extends {
+            (...args: infer A1): infer R1
+            (...args: infer A2): infer R2
+            (...args: infer A3): infer R3
+            (...args: infer A4): infer R4
+            (...args: infer A5): infer R5
+            (...args: infer A6): infer R6
+            (...args: infer A7): infer R7
+          }
+        ? [
+            (...args: A1) => R1,
+            (...args: A2) => R2,
+            (...args: A3) => R3,
+            (...args: A4) => R4,
+            (...args: A5) => R5,
+            (...args: A6) => R6,
+            (...args: A7) => R7,
+          ]
+        : FunctionType extends {
+              (...args: infer A1): infer R1
+              (...args: infer A2): infer R2
+              (...args: infer A3): infer R3
+              (...args: infer A4): infer R4
+              (...args: infer A5): infer R5
+              (...args: infer A6): infer R6
+            }
+          ? [
+              (...args: A1) => R1,
+              (...args: A2) => R2,
+              (...args: A3) => R3,
+              (...args: A4) => R4,
+              (...args: A5) => R5,
+              (...args: A6) => R6,
+            ]
+          : FunctionType extends {
+                (...args: infer A1): infer R1
+                (...args: infer A2): infer R2
+                (...args: infer A3): infer R3
+                (...args: infer A4): infer R4
+                (...args: infer A5): infer R5
+              }
+            ? [(...args: A1) => R1, (...args: A2) => R2, (...args: A3) => R3, (...args: A4) => R4, (...args: A5) => R5]
+            : FunctionType extends {
+                  (...args: infer A1): infer R1
+                  (...args: infer A2): infer R2
+                  (...args: infer A3): infer R3
+                  (...args: infer A4): infer R4
+                }
+              ? [(...args: A1) => R1, (...args: A2) => R2, (...args: A3) => R3, (...args: A4) => R4]
+              : FunctionType extends {
+                    (...args: infer A1): infer R1
+                    (...args: infer A2): infer R2
+                    (...args: infer A3): infer R3
+                  }
+                ? [(...args: A1) => R1, (...args: A2) => R2, (...args: A3) => R3]
+                : FunctionType extends {
+                      (...args: infer A1): infer R1
+                      (...args: infer A2): infer R2
+                    }
+                  ? [(...args: A1) => R1, (...args: A2) => R2]
+                  : FunctionType extends (...args: infer A1) => infer R1
+                    ? [(...args: A1) => R1]
+                    : any
+
+/**
+ * Extracts a tuple of parameter types from each overload
+ * of a given function type.
+ *
+ * @template FunctionType - The function type to extract parameter types from.
+ */
+export type OverloadedParameters<FunctionType> =
```

Comment:
Can we add a "unit test" to this? i.e. something in `types.test.ts` or a new file that shows what this util actually does